### PR TITLE
Document pivot and unpivot operators

### PR DESCRIPTION
**Pull request by ektravel**
### Description

This PR adds documentation for the PIVOT and UNPIVOT operators.

This PR has:

- [x] been self-reviewed.

### CODE REVIEW

File: docs/querying/sql.md [hunk 1]
```diff
@@ -85,6 +87,121 @@ FROM clause, metadata tables are not considered datasources. They exist only in
...
+```sql
+PIVOT (aggregation_function(column_to_aggregate)
+  FOR [column_with_values_to_pivot]
+  IN ( [pivoted_column1], [pivoted_column2] ...)
+)
+```
+
+PIVOT syntax parameters:
+
+* `aggregation_function`: An aggregation function such as SUM, COUNT, MIN, MAX, or AVG.
```
Comment (317brian) [inline docs/querying/sql.md:109]:
```suggestion
* `aggregation_function`: An aggregation function, such as SUM, COUNT, MIN, MAX, or AVG.
```

Comment (317brian) [reply docs/querying/sql.md:109]: Also the Google style guide says that the first letter after a colon isn't capitalized

File: docs/querying/sql.md [hunk 2]
```diff
@@ -85,6 +87,121 @@ FROM clause, metadata tables are not considered datasources. They exist only in
...
+## UNPIVOT
+
+:::info
+The UNPIVOT operator is an [experimental feature](../development/experimental-features.md).
+:::
+
+The UNPIVOT operator transforms existing column values into rows.
+Note that UNPIVOT isn't the exact reverse operation of PIVOT. The PIVOT operator carries out an aggregation and merges rows as needed. UNPIVOT doesn't reproduce the original rows that have been merged.
+
+The following is the general syntax for the UNPIVOT operator. Note that the UNPIVOT operator is enclosed in parentheses and forms part of the FORM clause of the query.
```
Comment (317brian) [inline docs/querying/sql.md:156]: typo `FORM`

File: website/.spelling [hunk 3]
```diff
@@ -778,6 +778,10 @@ APPROX_COUNT_DISTINCT_DS_THETA
 APPROX_QUANTILE_DS
 DS_QUANTILES_SKETCH
 APPROX_QUANTILE_FIXED_BUCKETS
+# Syntex
```
Comment (317brian) [inline website/.spelling:781]: what's # Syntex?

**Review by 317brian — CHANGES_REQUESTED** [review]
Some minor comments about the writing. It also needs to be added to https://github.com/ektravel/druid/blob/docs-pivot-unpivot/docs/querying/sql-functions.md

Will leave the technical review to someone more qualified

File: website/.spelling (hunk 3 above)
Comment (ektravel) [reply website/.spelling:781]: This is a typo. It was supposed to say "#syntax". I changed it to "#operators".

File: docs/querying/sql.md (hunk 2 above)
Comment (ektravel) [reply docs/querying/sql.md:156]: Good catch. Fixed.

File: docs/querying/sql.md (hunk 1 above)
Comment (ektravel) [reply docs/querying/sql.md:109]: In the Druid docs, most of description lists with bulleted run-in heading start text that follows colons with a capital letter. I can propose an exception to the style guide.

**Comment by ektravel** [comment]
`PIVOT` and `UNPIVOT` are relational operators. I will add them to https://github.com/apache/druid/blob/master/docs/querying/sql-operators.md instead of https://github.com/ektravel/druid/blob/docs-pivot-unpivot/docs/querying/sql-functions.md.

File: website/.spelling [hunk 4]
```diff
@@ -778,6 +778,10 @@ APPROX_COUNT_DISTINCT_DS_THETA
 APPROX_QUANTILE_DS
 DS_QUANTILES_SKETCH
 APPROX_QUANTILE_FIXED_BUCKETS
+# Operators
+pivoted
+UNPIVOT
```
Comment (soumyava) [inline website/.spelling:783]: should PIVOT be included here as well ?

**Review by soumyava — COMMENTED** [review]
Mostly LGTM

File: website/.spelling (hunk 4 above)
Comment (ektravel) [reply website/.spelling:783]: It didn't fail any spelling tests, so I don't think so.

**Review by soumyava — APPROVED** [review]
LGTM

File: docs/querying/sql.md [hunk 5]
```diff
@@ -85,6 +87,121 @@ FROM clause, metadata tables are not considered datasources. They exist only in
...
+|`#ar.wikipedia`|156.202.76.160|deleted|0|
+|`#ar.wikipedia`|156.212.124.165|added|451|
+|`#ar.wikipedia`|156.212.124.165|deleted|0|
+|`#ar.wikipedia`|160.166.147.167|added|1|
+|`#ar.wikipedia`|160.166.147.167|deleted|0|
+|`#ar.wikipedia`|185.99.32.50|added|1|
+|`#ar.wikipedia`|185.99.32.50|deleted|0|
+|`#ar.wikipedia`|197.18.109.148|added|0|
+|`#ar.wikipedia`|197.18.109.148|deleted|24|
+|`#ar.wikipedia`|2001:16A2:3C7:6C00:917E:AD28:FAD3:FD5C|added|1|
```
Comment (ektravel) [inline docs/querying/sql.md:199]:
```suggestion
|`#ar.wikipedia`|`2001:16A2:3C7:6C00:917E:AD28:FAD3:FD5C`|added|1|
```

File: docs/querying/sql.md [hunk 6]
```diff
@@ -85,6 +87,121 @@ FROM clause, metadata tables are not considered datasources. They exist only in
...
+|`#ar.wikipedia`|2001:16A2:3C7:6C00:917E:AD28:FAD3:FD5C|deleted|0|
```
Comment (ektravel) [inline docs/querying/sql.md:200]:
```suggestion
|`#ar.wikipedia`|`2001:16A2:3C7:6C00:917E:AD28:FAD3:FD5C`|deleted|0|
```

File: website/.spelling [hunk 7]
```diff
@@ -778,6 +778,12 @@ APPROX_COUNT_DISTINCT_DS_THETA
...
+unpivoted
+2001:16A2:3C7:6C00:917E:AD28:FAD3:FD5C
```
Comment (ektravel) [inline website/.spelling:785]:
```suggestion

```

**Review by 317brian — APPROVED** [review]
LGTM!

**Comment by 317brian** [comment]
@ektravel can you make a backport PR to 29.0 too? Thanks

**Pull request merged**